### PR TITLE
002 view job role info

### PR DESCRIPTION
**Pull request by AZZohaib**

### CODE REVIEW

File: src/controllers/JobRoleController.ts [hunk 1]
```diff
@@ -11,5 +11,18 @@ export const getAllJobRoles = async (req: express.Request, res: express.Response
...
-    
-}
\ No newline at end of file
+}
+
+export const getSingleJobRole = async (req: express.Request, res: express.Response): Promise<void> => {
+    try {
+        const jobRole = await getJobRoleById(req.params.id);
+        
+        // This is why I love JavaScript
```
Comment (JennicaMistry) [inline src/controllers/JobRoleController.ts:20]: Remove the comment

File: src/routes/UserRouter.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
-import express from 'express';
-import { getAllJobRoles } from '../controllers/JobRoleController';
+import express from "express";
+import { getSingleJobRole, getAllJobRoles } from "../controllers/JobRoleController";
 
 export const userRouter = express.Router();
 
+/* eslint-disable-next-line @typescript-eslint/no-misused-promises */
```
Comment (JennicaMistry) [inline src/routes/UserRouter.ts:6]: Disable eslint for the entire file, not just each route

File: src/services/JobRoleService.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
+import axios, {AxiosResponse} from "axios"
+import { JobRole } from "../models/JobRole"
```
Comment (JennicaMistry) [inline src/services/JobRoleService.ts:2]: Import as a type to save the polar bears

File: src/services/JobRoleService.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
+import axios, {AxiosResponse} from "axios"
```
Comment (JennicaMistry) [inline src/services/JobRoleService.ts:1]: Import as a type to save the polar bears

File: test/unit/services/JobRoleServiceTests.ts [hunk 5]
```diff
@@ -19,6 +20,19 @@ const jobRoleResponse: JobRoleResponse = {
     formattedLocations: "Birmingham"
 }
 
+const getJobRoleByIdResponse: JobRole = {
+  roleId: 5,
+  roleName: "Technical Architect",
+  description: "Test description for the role",
+  linkToJobSpec: "examplelink.co.uk",
+  responsibilities: "One, two, three, ninety-nine",
+  locations: Location.Birmingham,
```
Comment (JennicaMistry) [inline test/unit/services/JobRoleServiceTests.ts:29]: Remove the pluralisation.

**Review by JennicaMistry — CHANGES_REQUESTED** [review]

File: src/controllers/JobRoleController.ts (hunk 1 above)
Comment (AZZohaib) [reply src/controllers/JobRoleController.ts:20]: Done
image

File: src/services/JobRoleService.ts (hunk 3 above)
Comment (AZZohaib) [reply src/services/JobRoleService.ts:2]: Done
image

File: src/services/JobRoleService.ts (hunk 4 above)
Comment (AZZohaib) [reply src/services/JobRoleService.ts:1]: Done 
image

File: test/unit/services/JobRoleServiceTests.ts (hunk 5 above)
Comment (AZZohaib) [reply test/unit/services/JobRoleServiceTests.ts:29]: locations is actually a string that returns multiple locations (Comment has been overridden by Tom)

**Review by JennicaMistry — DISMISSED** [review]

**Review by Adnanhoq — DISMISSED** [review]

**Review by Adnanhoq — DISMISSED** [review]

**Review by DylanS-03 — APPROVED** [review]

**Review by JennicaMistry — DISMISSED** [review]

**Review by Adnanhoq — APPROVED** [review]

**Review by JennicaMistry — APPROVED** [review]

**Pull request merged**